### PR TITLE
[homekit] Synthesize Thermostat.TargetTemperature in some cases

### PR DESCRIPTION
**Pull request by ccutrer**
If HeatingThresholdTemperature or CoolingThresholdTemperature are provided, TargetTemperature doesn't have to be. in fact, TargetTemperature may not behave as your actual thermostat behaves. So document how HomeKit expects it, and then implement a synthesis of that behavior if your thermostat simply does not have it.

### CODE REVIEW

File: bundles/org.openhab.io.homekit/README.md [hunk 1]
```diff
@@ -520,6 +524,17 @@ Number          thermostat_cool_thrs       "Thermostat Cool Threshold Temp [%.1f
 Number          thermostat_heat_thrs       "Thermostat Heat Threshold Temp [%.1f °C]" (gThermostat) {homekit = "HeatingThresholdTemperature"}
 ```
 
+If your thermostat simply has a heating set point and cooling set point, and uses those points regardless of HEAT, COOL, or AUTO mode, you may simply omit the TargetTemperature characteristic, and the add-on will dynamically route to those characteristics as appropriate:
+
+```xtend
```
Comment (lsiepel) [inline bundles/org.openhab.io.homekit/README.md:529]: Please use java for item code fences:
For reference: https://github.com/openhab/openhab-docs/issues/999

**Review by lsiepel — CHANGES_REQUESTED** [review]
Very minor, otherwise LGTM. Would be nice for @yfre to confirm.

**Review by lsiepel — APPROVED** [review]

**Comment by ccutrer** [comment]
ping @yfre

**Comment by grzegorz914** [comment]
Looks like if we use all operations mode include AUTO  and the heating and cooling threshold is not set then the accessory is not published 

```
2024-07-23 08:17:19.992 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Biuro_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Biuro HVAC, Category=heating, Tags=[HVAC], Groups=[Biuro])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.055 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Kuchnia_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Kuchnia HVAC, Category=heating, Tags=[HVAC], Groups=[Kuchnia])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.063 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Jadalnia_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Jadalnia HVAC, Category=radiator, Tags=[HVAC], Groups=[Jadalnia])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.075 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:PrzedpokojDuzy_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Przedpokój duży HVAC, Category=radiator, Tags=[HVAC], Groups=[PrzedpokojDuzy])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.122 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Cwu (Type=GroupItem, Members=16, State=NULL, Label=C.W.U., Category=faucet, Tags=[Equipment], Groups=[Dom])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.138 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Sypialnia_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Sypialnia HVAC, Category=heating, Tags=[HVAC], Groups=[Sypialnia])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.178 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:LazienkaPietro_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Łazienka piętro HVAC, Category=heating, Tags=[HVAC], Groups=[LazienkaPietro])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.191 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Adam_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Adam HVAC, Category=heating, Tags=[HVAC], Groups=[Adam])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.280 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Kamila_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Kamila HVAC, Category=heating, Tags=[HVAC], Groups=[Kamila])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.286 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Garderoba_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Garderoba HVAC, Category=heating, Tags=[HVAC], Groups=[Garderoba])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.326 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:PrzedpokojMaly_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Przedpokój mały HVAC, Category=radiator, Tags=[HVAC], Groups=[PrzedpokojMaly])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.417 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:Salon_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Salon HVAC, Category=radiator, Tags=[HVAC], Groups=[Salon])  HomeKit type: 'Thermostat' characteristic: 'Empty'
2024-07-23 08:17:20.435 [WARN ] [mekit.internal.HomekitChangeListener] - Cannot create accessory Item:LazienkaParter_Hvac (Type=GroupItem, Members=6, State=NULL, Label=Łazienka parter HVAC, Category=heating, Tags=[HVAC], Groups=[LazienkaParter])  HomeKit type: 'Thermostat' characteristic: 'Empty' ```

**Pull request merged**